### ISSUE TITLE
invalid escapes in tab completion of paths with spaces

### ISSUE THREAD

**Issue by stevengj** [JuliaLang/julia#34701]
The REPL's pathname tab completion inserts backslashes before spaces in paths, which results in an invalid string because `\ ` is not a valid escape.  For example, if you create a file named `"foo bar"`
```jl
julia> touch("foo bar")
"foo bar"
```
and then type `"foo<TAB>` in the REPL, you get:
```jl
julia> "foo\ bar"  
ERROR: syntax: invalid escape sequence
Stacktrace:
 [1] top-level scope at REPL[1]:0
```
These backslashes seem very shell-like — what are they doing in Julia filenames?

**Comment by stevengj** [comment]
What a coincidence!  It looks like this was *just* fixed by @pfitzseb in #33816.

**Comment by stevengj** [comment]
Duplicate of #29304 anyway.
